src/index.tsx

```diff
@@ -148,6 +163,39 @@ class AzureAD extends React.Component<IProps, IState> {
       });
   }
 
+  private checkIfUserAuthenticated() {
+    if (this.state.authenticationState === AuthenticationState.Unauthenticated && this.isLoggedIn()) {
+      const idToken = this.getCacheItem(this.clientApplication.cacheLocation, 'msal.idtoken');
+      this.acquireTokens(idToken!);
+    }
+  }
+
+  // a person is logged in if UserAgentApplication has a current user, if there is an idtoken in the cache, and if the token in the cache is not expired
+  private isLoggedIn = () => {
+    const potentialLoggedInUser = this.clientApplication.getUser();
+    if (potentialLoggedInUser) {
+      const idToken = this.getCacheItem(this.clientApplication.cacheLocation, 'msal.idtoken');
+      const oldIDToken  = potentialLoggedInUser.idToken as any;
+      if (oldIDToken.exp && idToken) {
+        const expirationInMs = oldIDToken.exp * 1000; // AD returns in seconds
+        if (Date.now() < expirationInMs) { // id token isn't expired
```

Comment:
Do we need to worry about time zones here? Does AAD always return to us the epoch time for the expiration?

Reply:
docs say the time is epoch time, so no we don't